rustls-mbedcrypto-provider/src/hmac.rs

```diff
@@ -93,3 +97,55 @@ impl MbedHmacContext {
         }
     }
 }
+
+/// A HMAC tag, stored as a value.
+#[derive(Clone)]
+pub(crate) struct Tag {
+    buf: [u8; Self::MAX_LEN],
+    used: usize,
+}
+
+impl Tag {
+    /// Build a tag by copying a byte slice.
+    ///
+    /// The slice can be up to [`Tag::MAX_LEN`] bytes in length.
+    pub(crate) fn new(bytes: &[u8]) -> Self {
+        let mut tag = Self { buf: [0u8; Self::MAX_LEN], used: bytes.len() };
+        tag.buf[..bytes.len()].copy_from_slice(bytes);
+        tag
+    }
+
+    /// Build a tag with given capacity.
+    ///
+    /// The slice can be up to [`Tag::MAX_LEN`] bytes in length.
+    pub(crate) fn with_capacity(capacity: usize) -> Self {
```

Comment:
Could it be called `with_size` or `with_len`? I feel `with_capacity` is a bit misleading.

Reply:
sure, will rename it